samples/snippets/src/main/java/pubsublite/SubscriberExample.java

```diff
@@ -68,7 +69,7 @@ public static void subscriberExample(
 
     MessageReceiver receiver =
         (PubsubMessage message, AckReplyConsumer consumer) -> {
-          System.out.println("Id : " + message.getMessageId());
+          System.out.println("Id : " + PublishMetadata.decode(message.getMessageId()));
```

Comment:
I think this is a good idea, and I'm glad you are exposing the partition through me message ID. I would prefer if we didn't parse this into a *Publish* metadata, since the message ID is not really only the concern of the publisher. What if we called this MessageId?

Then we'd have:

`MessageId.decode(message.getMessageId());`

Reply:
Discussed offline, renaming this type will probably have to be done separately since its public. We'll discuss renaming out of band, and can do so at a later date.